x-pack/plugins/ml/public/data_frame/pages/data_frame_new_pivot/directive.tsx

```diff
@@ -11,27 +11,45 @@ import ReactDOM from 'react-dom';
 import { uiModules } from 'ui/modules';
 const module = uiModules.get('apps/ml', ['react']);
 
+import { StaticIndexPattern } from 'ui/index_patterns';
 import { I18nContext } from 'ui/i18n';
+import { IPrivate } from 'ui/private';
 
 // @ts-ignore
 import { SearchItemsProvider } from '../../../jobs/new_job/utils/new_job_utils';
+// Simple drop-in type until new_job_utils offers types.
+type CreateSearchItems = () => { indexPattern: StaticIndexPattern };
 
-import { IndexPatternContext } from '../../common';
+import { KibanaContext } from '../../common';
 import { Page } from './page';
 
-module.directive('mlNewDataFrame', ($route: any, Private: any) => {
+interface InjectorService {
```

Comment:
i'd imagine this will be useful for other ts directives we write, could it be moved to a common location.

Reply:
Good idea, moved to `common/types/angular.ts` in 19476ee.